### PR TITLE
optimize the makefile process

### PR DESCRIPTION
**Pull request by bitsf**
Change-Id: Ibe806972a19cd69bfd90be051cdc340c4d7c6afb
Signed-off-by: Ziming Zhang <zziming@vmware.com>

### CODE REVIEW

**Comment by reasonerjt** [comment]
Could we add more meaningful messages to the commit?

File: Makefile [hunk 1]
```diff
@@ -158,23 +160,24 @@ GOIMAGEBUILDCMD=/usr/local/go/bin/go build -mod vendor
...
+GOBUILDPATH_MIGRATEPATCH=$(GOBUILDPATHINCONTAINER)/src/cmd/migrate-patch
+GOBUILDMAKEPATH=make
 GOBUILDMAKEPATH_CORE=$(GOBUILDMAKEPATH)/photon/core
 GOBUILDMAKEPATH_JOBSERVICE=$(GOBUILDMAKEPATH)/photon/jobservice
 GOBUILDMAKEPATH_REGISTRYCTL=$(GOBUILDMAKEPATH)/photon/registryctl
 GOBUILDMAKEPATH_NOTARY=$(GOBUILDMAKEPATH)/photon/notary
 
 # binary
-CORE_BINARYPATH=$(MAKEDEVPATH)/core
+CORE_BINARYPATH=$(BUILDPATH)/$(GOBUILDMAKEPATH_CORE)
```
Comment (wy65701436) [inline Makefile]: remove `MAKEDEVPATH` also?

File: Makefile [hunk 2]
```diff
@@ -254,14 +257,14 @@ PACKAGE_OFFLINE_PARA=-zcvf harbor-offline-installer-$(PKGVERSIONTAG).tgz \
 					$(HARBORPKG)/prepare \
 					$(HARBORPKG)/LICENSE $(HARBORPKG)/install.sh \
 					$(HARBORPKG)/common.sh \
-					$(HARBORPKG)/harbor.yml
+					$(HARBORPKG)/harbor.yml.tmpl
```
Comment (wy65701436) [inline Makefile]: we need to add the `harbor.yml.tmpl` existence validation in `prepare`

Comment (bitsf) [reply Makefile]: added check for harbor.yml in prepare file

File: Makefile (hunk 1 above)
Comment (bitsf) [reply Makefile]: removed

File: Makefile [hunk 3]
```diff
@@ -64,13 +64,13 @@
...
 BUILDPATH=$(CURDIR)
 MAKEPATH=$(BUILDPATH)/make
-MAKEDEVPATH=$(MAKEPATH)/dev
 MAKE_PREPARE_PATH=$(MAKEPATH)/photon/prepare
 SRCPATH=./src
 TOOLSPATH=$(BUILDPATH)/tools
 CORE_PATH=$(BUILDPATH)/src/core
 PORTAL_PATH=$(BUILDPATH)/src/portal
 CHECKENVCMD=checkenv.sh
+OSNAME=$(shell uname)
```
Comment (wy65701436) [inline Makefile]: I don't see where is used for?

File: make/photon/prepare/utils/nginx.py [hunk 4]
```diff
@@ -59,6 +59,7 @@ def render_nginx_template(config_dict):
             nginx_conf,
             uid=DEFAULT_UID,
             gid=DEFAULT_GID,
+            https_redirect='$host' + ('https_port' in config_dict and (":" + str(config_dict['https_port'])) or ""),
```
Comment (ninjadq) [inline make/photon/prepare/utils/nginx.py]: why add https code here?

File: make/photon/prepare/templates/nginx/nginx.https.conf.jinja [hunk 5]
```diff
@@ -162,6 +162,6 @@ http {
   server {
       listen 8080;
       #server_name harbordomain.com;
-      return 308 https://$host$request_uri;
+      return 308 https://{{https_redirect}}$request_uri;
```
Comment (ninjadq) [inline make/photon/prepare/templates/nginx/nginx.https.conf.jinja]: why add https code here?

File: Makefile (hunk 3 above)
Comment (bitsf) [reply Makefile]: I intent to fix "sed -i" on MacOS with this env, but finally not use it.
Removed it

**Pull request merged**